nucliadb_vectors/src/data_types/data_store.rs

```diff
@@ -89,6 +89,37 @@ pub fn get_value<I: Interpreter>(interpreter: I, src: &[u8], id: usize) -> &[u8]
     interpreter.read_exact(&src[pointer..]).0
 }
 
+use lazy_static::lazy_static;
+use libc;
+
+#[cfg(not(target_os = "windows"))]
+pub fn will_need(src: &[u8], id: usize, dimension: usize) {
+    lazy_static! {
+        static ref PAGE_SIZE: usize = unsafe { libc::sysconf(libc::_SC_PAGESIZE) } as usize;
+    };
+
+    // Calculate node struct size without reading anything, so make some estimates
+    // Will only load up to the vectors section, it ignores the key/labels because
+    // they are harder to estimate and less useful (only when filtering, similarity is always used)
+    let metadata_len = 16; // Estimate
```

Comment:
For this use, it might be simpler to just reorder the store so vectors are first or separated from other stuff like metadata, key, labels.

Anyway, enforcing a metadata limit now might break indexing resources with longer metadata (and reindexing existing resources during a rollover), so we would have to calculate the actual used maximum. I don't think it's worth it to do this now since this is just an optimization.